config.yaml

```diff
@@ -64,3 +64,69 @@ sdk:
+  # Configure the meter provider.
+  meter_provider:
+    # Metric exporters. Each exporter key refers to the type of the exporter. Values configure the exporter. Exporters must be associated with a metric reader.
+    exporters:
+      # Configure the otlp exporter.
+      otlp:
+        # Sets the protocol.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_PROTOCOL, OTEL_EXPORTER_OTLP_METRICS_PROTOCOL
+        protocol: http/protobuf
+        # Sets the endpoint.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_ENDPOINT, OTEL_EXPORTER_OTLP_METRICS_ENDPOINT
+        endpoint: http://localhost:4317/v1/metrics
+        # Sets the certificate.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_CERTIFICATE, OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE
+        certificate: /app/cert.pem
+        # Sets the mTLS private client key.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_CLIENT_KEY, OTEL_EXPORTER_OTLP_METRICS_CLIENT_KEY
+        client_key: /app/cert.pem
+        # Sets the mTLS client certificate.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE, OTEL_EXPORTER_OTLP_METRICS_CLIENT_CERTIFICATE
+        client_certificate: /app/cert.pem
+        # Sets the headers.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_HEADERS, OTEL_EXPORTER_OTLP_METRICS_HEADERS
+        headers:
+          api-key: 1234
+        # Sets the compression.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_COMPRESSION, OTEL_EXPORTER_OTLP_METRICS_COMPRESSION
+        compression: gzip
+        # Sets the max time to wait for each export.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_TIMEOUT, OTEL_EXPORTER_OTLP_METRICS_TIMEOUT
+        timeout: 10000
+        # Sets the temporality preference.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_METRICS_TEMPORALITY_PREFERENCE
+        temporality_preference: delta
+        # Sets the default histogram aggregation.
+        #
+        # Environment variable: OTEL_EXPORTER_OTLP_METRICS_DEFAULT_HISTOGRAM_AGGREGATION
+        default_histogram_aggregation: exponential_bucket_histogram
+    # List of metric readers. Each metric reader has a name and args used to configure it.
+    metric_readers:
+      # Add a periodic metric reader.
+      #
+      # Environment variable: OTEL_METRICS_EXPORT_*, OTEL_METRICS_EXPORTER
+      - name: periodic
+        args:
+          # Sets delay interval between the start of two consecutive export attempts.
+          #
+          # Environment variable: OTEL_METRIC_EXPORT_INTERVAL
+          interval: 5000
+          # Sets the maximum allowed time to export data.
+          #
+          # Environment variable: OTEL_METRIC_EXPORT_TIMEOUT
+          timeout: 30000
+          # Sets the exporter. Exporter must refer to a key in sdk.meter_provider.exporters.
+          #
+          # Environment variable: OTEL_METRICS_EXPORTER
+          exporter: otlp
```

Comment:
What does this look like for the pull-based exporter which models the MetricReader?

Reply:
That's a good question. And I think its complicated by the fact that the SDK spec allows prometheus to be implemented as either a reader or a pull based exporter.

Let's talk about the prometheus example. I'm biased by the java implementation in which the prometheus exporter is just another reader, but here's what I would do:
```
sdk:
  meter_provider:
    metric_readers:
      - name: prometheus
         args:
           port: 5555
```
The alternative would be to list prometheus under exporters, but then which reader would it be associated with?

Perhaps for the purposes of file based SDK configuration, pull based exporters should just be entires under `metric_readers`. I.e. no entry needed under `sdk.meter_provider.exporters`.

Reply:
This makes sense to me, i would add the prometheus metric reader as an example here, since it wouldn't require an exporter parameter. I guess this means the schema would have to be validated differently per type of metric reader, which i guess is no different than validating processors differently based on their types.

Reply:
Isn't that going to be a problem for SDKs ([cpp](https://opentelemetry.io/docs/instrumentation/cpp/exporters/#prometheus) for example ) that chose not to implement the Prometheus exporter as a metric reader? It wouldn't expect a metric reader named `prometheus`.

Reply:
@srikanthccv curious how that works in cpp. [This](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#pull-metric-exporter) diagram shows options for implementing pull exporters. If cpp has prometheus modeled as a metric exporter, which metric reader is it associated with?

Reply:
It is paired with the periodic reader. While the exporter as metric reader implementation invokes the `collect` on demand as `prometheus.Gatherer` is run, CPP implementation adds the metrics to [`Collector`](https://prometheus.io/docs/instrumenting/writing_exporters/#collectors) for each collection interval configured on the periodic metric reader.  It's hard to define config for this given the flexibility spec has given to SDKs. Others may have taken a different approach. This is probably worth discussing in a separate issue.

Reply:
Gotcha. Thanks, and I agree it will be hard to define the config that makes everyone happy given the flexibility in the spec. There will have to be a compromise somewhere. Opened #24 to track this.